Fix hours of operation

lmillet3804 opened the pull request:
-added three util methods to handle parsing of hours
- disable (and gray out) restaurant button if it is 30 minutes to closing
- give a warning message if time selected is outside hours of operation

lmillet3804 committed:
updated frontend for hours of operation:

-added three util methods to handle parsing of hours
- disable (and gray out) restaurant button if it is 30 minutes to closing
- give a warning message if time selected is outside hours of operation

TheGoldenPatrik1 commented on app/buy/index.js:
```diff
@@ -95,13 +114,25 @@ const Buy = () => {
-                                style={styles.locationLogo}
-                                resizeMode="contain"
-                            />
+                            {isDisabled(option.hours) ? (
```
Can we show a message somewhere that tells them that restaurants are disabled if they're within 30 minutes of closing? This could just be a general message at the top or bottom of the page in small text that only shows up if at least one of the restaurants is disabled; I've heard that solely disabling buttons is confusing to users

TheGoldenPatrik1 commented on app/buy/pickTime.js:
```diff
@@ -35,13 +43,34 @@ const PickTime = () => {
         }
     }, [orderID])
 
+    useEffect(() => {
```
Don't we want the onConfirm event to not go through successfully if warningText has length > 0? Or are you leaving open the possibility of ordering in the case that our hours of operation are inaccurate?

lmillet3804 replied:
Yes, the latter. Is that a good idea or should I change it?

TheGoldenPatrik1 replied:
Arden said the status quo is fine. And I agree that I don't trust our hours of operation to always be accurate

TheGoldenPatrik1 commented on app/buy/pickTime.js:
```diff
@@ -35,13 +43,34 @@ const PickTime = () => {
+        const openTime = convertTimeToDateObject(openTimeString)
+        if (dt < openTime || dt > closeTime) {
+            setWarningText(
+                "Warning: the time you have selected is outside the restaurant's normal hours of operation."
```
This reminded me that we should probably display the hours of operation somewhere. Definitely here, but maybe also elsewhere in the buy process? Think about it & add them somewhere (putting them on the main /buy page would be huge, if you could get them in small text just under each image)

lmillet3804 committed:
addressed PR comments:

-added some hours info underneath restaurant images
-added a message if at least one restaurant is disabled
-added current day's hours of operation in pickTime screen
-fixed a glitch with how times were being converted
